compiler/src/yul/runtime/abi.rs

```diff
@@ -1,22 +1,16 @@
 use crate::abi::utils as abi_utils;
 use crate::errors::CompileError;
-use crate::yul::namespace::scopes::ContractDef;
-#[allow(unused_imports)]
-use crate::yul::namespace::types::{
-    Base,
-    FixedSize,
-};
-use std::collections::HashMap;
+#[cfg(test)]
+use fe_semantics::namespace::types::Base;
+use fe_semantics::namespace::types::FixedSize;
+use fe_semantics::FunctionAttributes;
 use yultsur::*;
 
 /// Builds a switch statement that dispatches calls to the contract.
-pub fn dispatcher(
-    interface: &[String],
-    defs: &HashMap<String, ContractDef>,
-) -> Result<yul::Statement, CompileError> {
-    let arms = interface
+pub fn dispatcher(attributes: Vec<FunctionAttributes>) -> Result<yul::Statement, CompileError> {
+    let arms = attributes
         .iter()
-        .map(|name| dispatch_arm(name.to_owned(), defs))
+        .map(|a| dispatch_arm(a.to_owned()))
```

Comment:
As a team convention we try to avoid single letter variable names as much as possible. (Except for life time annotations and generic type params)

Reply:
I can update these.